index.js

```diff
@@ -457,6 +457,50 @@ class Corestore extends ReadyResource {
     }
   }
 
+  async staticify(core, opts) {
+    if (!this.opened) await this.ready()
+    if (!core.opened) await core.ready()
+
+    const rx = core.state.storage.read()
+
+    const headPromise = rx.getHead()
+    const authPromise = rx.getAuth()
+
+    rx.tryFlush()
+
+    const [head, auth] = await Promise.all([headPromise, authPromise])
+    if (!head || head.length === 0) throw new Error('Core must have data')
+
+    const prologue = {
+      length: head.length,
+      hash: head.rootHash
```

Comment:
Sanity check: head.rootHash and head.length are guaranteed to exist?

Reply:
it must have a hash if length > 0, yes